src/value_array.rs

```diff
@@ -28,8 +28,7 @@ impl ValueArray {
         }
     }
 
-    pub fn append<'a, P: Into<Option<&'a Value>>>(&mut self, value: P) {
-        let value = value.into();
+    pub fn append(&mut self, value: Option<&Value>) {
```

Comment:
This should not be `nullable`. While the C API allows it, it is unsafe. Same for the other functions below